src/allmydata/web/common.py

```diff
@@ -260,29 +266,43 @@ def humanize_failure(f):
                  "writecap in the write slot if desired, would also work in this "
                  "case.") % quoted_name
         return (t, http.BAD_REQUEST)
-    if f.check(MustBeDeepImmutableError):
-        quoted_name = quote_output(f.value.args[1], encoding="utf-8")
+    if isinstance(exc, MustBeDeepImmutableError):
+        quoted_name = quote_output(exc.args[1], encoding="utf-8")
         t = ("MustBeDeepImmutableError: a cap passed to this operation for "
              "the child named %s, needed to be immutable but was not. Either "
              "the cap is being added to an immutable directory, or it was "
              "originally retrieved from an immutable directory as an unknown "
              "cap.") % quoted_name
         return (t, http.BAD_REQUEST)
-    if f.check(MustBeReadonlyError):
-        quoted_name = quote_output(f.value.args[1], encoding="utf-8")
+    if isinstance(exc, MustBeReadonlyError):
+        quoted_name = quote_output(exc.args[1], encoding="utf-8")
         t = ("MustBeReadonlyError: a cap passed to this operation for "
              "the child named '%s', needed to be read-only but was not. "
              "The cap is being passed in a read slot (ro_uri), or was retrieved "
              "from a read slot as an unknown cap.") % quoted_name
         return (t, http.BAD_REQUEST)
-    if f.check(blacklist.FileProhibited):
-        t = "Access Prohibited: %s" % quote_output(f.value.reason, encoding="utf-8", quotemarks=False)
+    if isinstance(exc, blacklist.FileProhibited):
+        t = "Access Prohibited: %s" % quote_output(exc.reason, encoding="utf-8", quotemarks=False)
         return (t, http.FORBIDDEN)
-    if f.check(WebError):
-        return (f.value.text, f.value.code)
-    if f.check(FileTooLargeError):
-        return (f.getTraceback(), http.REQUEST_ENTITY_TOO_LARGE)
-    return (str(f), None)
+    if isinstance(exc, WebError):
+        return (exc.text, exc.code)
+    if isinstance(exc, FileTooLargeError):
+        return ("FileTooLargeError: %s" % (exc,), http.REQUEST_ENTITY_TOO_LARGE)
+    return (str(exc), None)
+
+
+def humanize_failure(f):
+    """
+    Create an human-oriented description of a failure along with some HTTP
+    metadata.
+
+    :param Failure f: The failure to describe.
+
+    :return (bytes, int): A tuple of some prose and an HTTP code describing
+        the failure.
+    """
+    return humanize_exception(f.value)
+
 
 class MyExceptionHandler(appserver.DefaultExceptionHandler, object):
```

Comment:
Just wanted to make sure that I understand changes to `MyExceptionHandler` and other changes in this PR.  My thinking is that:

* `MyExceptionHandler` will go away when we finally replace `nevow.appserver.NevowSite` with `twisted.web.server.Site`, since there's no equivalent of installing such a thing in Twisted.
* Because there's no equivalent of installing a `Site`-wide exception page renderer, the `@render_exception` decorator would be doing approximately what `MyExceptionHandler` was doing for `NevowSite`.

Is this correct?

Reply:
> MyExceptionHandler will go away when we finally replace nevow.appserver.NevowSite with twisted.web.server.Site, since there's no equivalent of installing such a thing in Twisted.

Indeed, there is no equivalent.  I think that means that `MyExceptionHandler` going away is the most logical path.  (The alternative would be contributing something equivalent to Twisted but Tahoe-LAFS can't depend on any new Twisted features until it is ported to Python 3 and it cannot be ported to Python 3 until the Nevow dependency is gone so ...)

> Because there's no equivalent of installing a Site-wide exception page renderer, the @render_exception decorator would be doing approximately what MyExceptionHandler was doing for NevowSite.

Right - as long as it gets applied individually to each render method.